src/services/paras/ParasService.ts

```diff
@@ -102,4 +121,255 @@ export class ParasService extends AbstractService {
+				hash,
+				paraId
+			);
+			onboardingAs = paraGenesisArgs.parachain.isTrue
+				? 'parachain'
+				: 'parathread';
+		}
+
+		return {
+			at,
+			paraLifeCycle,
+			onboardingAs,
+			leases: leasesFormatted,
+		};
+	}
+
+	/**
+	 * @param hash `BlockHash` to make call at
+	 * @returns information on the current auction. Most fields will be null if
+	 * if there is no ongoing auction.
+	 */
+	async auctionsCurrent(hash: BlockHash): Promise<IAuctionsCurrent> {
+		const [auctionInfoOpt, { number }, auctionCounter] = await Promise.all([
+			this.api.query.auctions.auctionInfo.at<Option<Vec<AbstractInt>>>(hash),
+			this.api.rpc.chain.getHeader(hash),
+			this.api.query.auctions.auctionCounter.at<AbstractInt>(hash),
+		]);
+		const blockNumber = number.unwrap();
+
+		const endingPeriod = this.api.consts.auctions.endingPeriod as AbstractInt;
+
+		let leasePeriodIndex: IOption<AbstractInt>,
+			beginEnd: IOption<AbstractInt>,
+			finishEnd: IOption<BN>,
+			phase: IOption<AuctionPhase>,
+			winning;
+		if (auctionInfoOpt.isSome) {
+			[leasePeriodIndex, beginEnd] = auctionInfoOpt.unwrap();
+			const endingOffset = this.endingOffset(blockNumber, beginEnd);
+			if (endingOffset) {
+				winning = await this.api.query.auctions.winning.at<Option<WinningData>>(
+					hash,
+					endingOffset
+				);
+			} else {
+				winning = null;
+			}
+
+			finishEnd = beginEnd.add(endingPeriod);
+			phase = beginEnd.gt(blockNumber) ? 'preEnding' : 'ending';
+		} else {
+			leasePeriodIndex = null;
+			beginEnd = null;
+			finishEnd = null;
+			phase = null;
+			winning = null;
+		}
+
+		const leasePeriods = isSome(leasePeriodIndex)
+			? Array(4)
+					.fill(0)
+					.map((_, i) => i + (leasePeriodIndex as BN).toNumber())
+			: null;
+
+		return {
+			at: {
+				hash,
+				height: blockNumber.toString(10),
+			},
+			beginEnd,
+			finishEnd,
+			phase,
+			// If there is no current auction, this will be the index of the previous auction
+			auctionIndex: auctionCounter,
+			leasePeriods,
+			winning,
+		};
+	}
+
+	/**
+	 * @param hash `BlockHash` to make call at
+	 * @param includeCurrentLeaseHolders wether or not to include the paraIds of
+	 * all the curent lease holders. Not including is likely faster and reduces
+	 * response size.
+	 * @returns general information about the current lease period
+	 */
+	async leasesCurrent(
+		hash: BlockHash,
+		includeCurrentLeaseHolders: boolean
+	): Promise<ILeasesCurrent> {
+		let blockNumber, currentLeaseHolders;
+		if (!includeCurrentLeaseHolders) {
+			const { number } = await this.api.rpc.chain.getHeader(hash);
+			blockNumber = number.unwrap();
+		} else {
+			const [{ number }, leaseEntries] = await Promise.all([
+				this.api.rpc.chain.getHeader(hash),
+				this.api.query.slots.leases.entriesAt<
+					Vec<Option<ITuple<[AccountId, BalanceOf]>>>,
+					[ParaId]
+				>(hash),
+			]);
+
+			blockNumber = number.unwrap();
+
+			currentLeaseHolders = leaseEntries
+				.filter(([_k, leases]) => leases[0].isSome)
```

Comment:
I like the use of `isSome`, I think it has the ability to also optimize code in a lot of other service, etc

Reply:
Yup - just to clarify though, this is `.isSome` is  a property on the polkadot-js `Option` enum which extends `Codec` - which is different than the `IOption` type alias we have defined within this repo.